cachi2/core/package_managers/gomod.py

```diff
@@ -136,7 +137,33 @@ def fetch_gomod_source(request: Request) -> RequestOutput:
     )
 
 
-def _find_missing_gomod_files(source_path: Path, subpaths: list[str]) -> list[Path]:
+def _protect_against_symlinks(app_dir: RootedPath) -> None:
+    """Try to prevent go subcommands from following suspicious symlinks.
+
+    The go command doesn't particularly care if the files it reads are subpaths of the directory
+    where it is executed. Check some of the common paths that the subcommands may read.
+    """
+
+    def check_potential_symlink(relative_path: Union[str, Path]) -> None:
+        try:
+            app_dir.join_within_root(relative_path)
+        except PathOutsideRoot:
+            raise PackageRejected(
+                reason=(
+                    f"Found suspicious symlink ({relative_path}) in Go module directory "
```

Comment:
I think we should explain briefly in this exception why the symlink is suspicious (i.e. points to ouside of the repo).

Reply:
I decided to raise the original PathOutsideRoot and just modify the "solution" to explain the symlink problem

Reply:
@brunoapimentel PTAL if it's better now

Reply:
Looks great :+1: